edx_oauth2_provider/forms.py

```diff
@@ -88,29 +87,41 @@ def clean(self):
             # verified_email, we can uncomment the following line.
             # or not user.is_active
         ):
-            raise OAuthValidationError({'error': 'invalid_grant'})
+            error_description = "Username does not exist or invalid credentials given for username '{}'.".format(username)
+            log.exception("OAuth2: {}".format(error_description))
+            raise OAuthValidationError({
+                'error': 'invalid_grant',
+                'error_description': error_description
+            })
 
         data['user'] = user
         return data
 
 
-class PublicPasswordGrantForm(PasswordGrantForm,
-                              provider.oauth2.forms.PublicPasswordGrantForm):
+class PublicPasswordGrantForm(PasswordGrantForm, provider.oauth2.forms.PublicPasswordGrantForm):
     """
     Form wrapper to ensure the the customized PasswordGrantForm is used
     during client authentication.
-
     """
     def clean(self):
         data = super(PublicPasswordGrantForm, self).clean()
 
         try:
             client = Client.objects.get(client_id=data.get('client_id'))
         except Client.DoesNotExist:
-            raise OAuthValidationError({'error': 'invalid_client'})
+            error_description = "Client ID '{}' does not exist.".format(data.get('client_id'))
+            log.exception("OAuth2: {}".format(error_description))
+            raise OAuthValidationError({
+                'error': 'invalid_client',
+                'error_description': error_description
+            })
 
         if client.client_type != provider.constants.PUBLIC:
-            raise OAuthValidationError({'error': 'invalid_client'})
-
+            error_description = "'{}' is not a public client.".format(client.client_type)
+            log.exception("OAuth2: {}".format(error_description))
```

Comment:
sorry for not being clear.  You can change it to `log.error`.  We probably don't need the stack trace for this anyway.